xarray/namedarray/core.py

```diff
@@ -703,6 +721,154 @@ def sizes(self) -> dict[_Dim, _IntOrUnknown]:
         """Ordered mapping from dimension names to lengths."""
         return dict(zip(self.dims, self.shape))
 
+    def chunk(
+        self,
+        chunks: (
+            int
+            | Literal["auto"]
+            | tuple[int, ...]
+            | tuple[tuple[int, ...], ...]
+            | Mapping[Any, None | int | tuple[int, ...]]
+        ) = {},
+        name: str | None = None,
+        lock: bool | None = None,
+        inline_array: bool | None = None,
+        chunked_array_type: str | ChunkManagerEntrypoint | None = None,
+        from_array_kwargs=None,
+        **chunks_kwargs: Any,
+    ) -> Self:
```

Comment:
Ccing @TomNicholas

Reply:
We can simply this a alot. It should be possible to remove `name`, `lock`, and `inline_array` in favour of them being passed via `from_array_kwargs`, because they are just dask-specific kwargs to `ChunkManager.from_array`. IIRC they already can be passed this way, they are just also explicitly present in the signature for backwards compatibility.

Reply:
Chunks should be positional, but I don't think any of the others need be.

Reply:
considering how large this PR has gotten, should we consider addressing these additional issues in separate PRs?

Reply:
never mind :) Tom pointed it out below: https://github.com/pydata/xarray/pull/8319#issuecomment-1781225864